frontend/packages/knative-plugin/src/components/add/event-sources/KafkaSourceSection.tsx

```diff
@@ -1,37 +1,103 @@
 import * as React from 'react';
-import { useFormikContext, FormikValues } from 'formik';
+import * as _ from 'lodash';
 import FormSection from '@console/dev-console/src/components/import/section/FormSection';
-import { InputField, TextColumnField } from '@console/shared';
+import { InputField, SelectInputField, SelectInputOption } from '@console/shared';
 import { TextInputTypes } from '@patternfly/react-core';
 import KafkaSourceNetSection from './KafkaSourceNetSection';
 import ServiceAccountDropdown from '../../dropdowns/ServiceAccountDropdown';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { useK8sWatchResources } from '@console/internal/components/utils/k8s-watch-hook';
+import { getBootstrapServers } from '../../../utils/create-eventsources-utils';
+import { strimziResourcesWatcher } from '../../../utils/get-knative-resources';
 
 const KafkaSourceSection: React.FC = () => {
-  const { values } = useFormikContext<FormikValues>();
-  const {
-    data: {
-      kafkasource: { bootstrapServers, topics },
-    },
-  } = values;
+  const [bootstrapServers, setBootstrapServers] = React.useState<SelectInputOption[]>([]);
+  const [bsPlaceholder, setBsPlaceholder] = React.useState<React.ReactNode>('');
+  const [kafkaTopics, setKafkaTopics] = React.useState<SelectInputOption[]>([]);
+  const [ktPlaceholder, setKtPlaceholder] = React.useState<React.ReactNode>('');
+
+  const memoResources = React.useMemo(() => strimziResourcesWatcher(), []);
+  const { kafkas, kafkatopics } = useK8sWatchResources<{
+    [key: string]: K8sResourceKind[];
+  }>(memoResources);
+
+  React.useEffect(() => {
```

Comment:
This approach works but in future, `React.useMemo` is better suited to these situations because you aren't executing side effects but rather computing values for the current render.

Reply:
Makes sense, I have updated it to use `React.useMemo` instead.